src/StrictifyFloat.cpp

```diff
@@ -3,82 +3,199 @@
+        if (op->a.type().is_float()) {
+            return mutate(op->b <= op->a);
+        } else {
+            return IRMutator::visit(op);
+        }
+    }
 
-        return IRMutator::visit(call);
+    Expr visit(const EQ *op) override {
+        return visit_binop(op, Call::strict_eq);
     }
 
-    using IRMutator::mutate;
+    Expr visit(const NE *op) override {
+        if (op->a.type().is_float()) {
+            return !mutate(op->b == op->a);
+        } else {
+            return IRMutator::visit(op);
+        }
+    }
 
-    Expr mutate(const Expr &expr) override {
-        if (!expr.defined()) {
-            return expr;
+    Expr visit(const Call *op) override {
+        if (op->call_type == Call::PureExtern &&
+            (op->name == "sqrt_f16" ||
+             op->name == "sqrt_f32" ||
+             op->name == "sqrt_f64")) {
+            return Call::make(op->type, Call::strict_sqrt,
+                              {mutate(op->args[0])}, Call::PureIntrinsic);
+        } else {
+            return IRMutator::visit(op);
         }
-        Expr e = IRMutator::mutate(expr);
-        if (e.type().is_float()) {
-            switch (strictness) {
-            case FastMath:
-                return e;
-            case StrictFloat:
-                return strict_float(e);
+    }
+};
+
+class Unstrictify : public IRMutator {
+    using IRMutator::visit;
+
+    Expr visit(const Call *op) override {
+        if (op->is_intrinsic(Call::strict_add)) {
+            return mutate(op->args[0] + op->args[1]);
+        } else if (op->is_intrinsic(Call::strict_sub)) {
+            return mutate(op->args[0] - op->args[1]);
+        } else if (op->is_intrinsic(Call::strict_mul)) {
+            return mutate(op->args[0] * op->args[1]);
+        } else if (op->is_intrinsic(Call::strict_div)) {
+            return mutate(op->args[0] / op->args[1]);
+        } else if (op->is_intrinsic(Call::strict_min)) {
+            return mutate(min(op->args[0], op->args[1]));
+        } else if (op->is_intrinsic(Call::strict_max)) {
+            return mutate(max(op->args[0], op->args[1]));
+        } else if (op->is_intrinsic(Call::strict_sqrt)) {
+            Expr e;
+            if (op->type.bits() == 64) {
+                e = Call::make(op->type, "sqrt_f64", op->args, Call::PureExtern);
+            } else if (op->type.bits() == 16) {
+                e = Call::make(op->type, "sqrt_f16", op->args, Call::PureExtern);
+            } else {
+                e = Call::make(op->type, "sqrt_f32", op->args, Call::PureExtern);
             }
+            return mutate(e);
+        } else if (op->is_intrinsic(Call::strict_lt)) {
+            return op->args[0] < op->args[1];
+        } else if (op->is_intrinsic(Call::strict_le)) {
+            return op->args[0] <= op->args[1];
+        } else if (op->is_intrinsic(Call::strict_eq)) {
+            return op->args[0] == op->args[1];
+        } else {
+            return IRMutator::visit(op);
```

Comment:
Recursing feels wrong (see comment above). Perhaps the recursion should be a ` bool` constructor argument to the mutator?